datadog-opentelemetry/src/metrics_reader.rs

```diff
@@ -0,0 +1,229 @@
+
+        #[cfg(not(feature = "metrics-grpc"))]
+        if matches!(protocol, OtlpProtocol::Grpc) {
+            dd_warn!("FEATURE MISMATCH: Protocol 'grpc' configured (OTEL_EXPORTER_OTLP_PROTOCOL or OTEL_EXPORTER_OTLP_METRICS_PROTOCOL) but 'metrics-grpc' feature is NOT enabled. Metrics will not be exported. Enable the 'metrics-grpc' feature in Cargo.toml or set OTEL_EXPORTER_OTLP_PROTOCOL=http/protobuf.");
+            return Ok(SdkMeterProvider::builder().build());
+        }
+
+        #[cfg(not(feature = "metrics-http"))]
+        if matches!(protocol, OtlpProtocol::HttpProtobuf) {
+            dd_warn!("FEATURE MISMATCH: Protocol 'http/protobuf' configured (OTEL_EXPORTER_OTLP_PROTOCOL or OTEL_EXPORTER_OTLP_METRICS_PROTOCOL) but 'metrics-http' feature is NOT enabled. Metrics will not be exported. Enable the 'metrics-http' feature in Cargo.toml or set OTEL_EXPORTER_OTLP_PROTOCOL=grpc.");
+            return Ok(SdkMeterProvider::builder().build());
+        }
+
+        if matches!(protocol, OtlpProtocol::HttpJson) {
+            dd_warn!("UNSUPPORTED PROTOCOL: HTTP/JSON protocol is not natively supported by opentelemetry-otlp. Metrics will not be exported. Use 'grpc' or 'http/protobuf' instead.");
+            return Ok(SdkMeterProvider::builder().build());
+        }
+
+        let mut endpoint = get_otlp_metrics_endpoint(&config, &protocol)?;
+
+        if matches!(protocol, OtlpProtocol::HttpProtobuf) && !endpoint.ends_with("/v1/metrics") {
+            endpoint = endpoint.trim_end_matches('/').to_string();
+            if !endpoint.is_empty() {
+                endpoint.push_str("/v1/metrics");
+            }
+        }
+
+        let temporality = match config
+            .otel_metrics_temporality_preference()
+            .to_lowercase()
+            .as_str()
+        {
+            "cumulative" => opentelemetry_sdk::metrics::Temporality::Cumulative,
+            _ => opentelemetry_sdk::metrics::Temporality::Delta,
+        };
+        let timeout = Duration::from_millis(get_otlp_metrics_timeout(&config) as u64);
+
+        let exporter = match build_exporter(protocol, endpoint.clone(), timeout, temporality) {
+            Ok(exporter) => exporter,
+            Err(err) => {
+                dd_warn!(
+                    "Failed to create metrics exporter: {}. Metrics will not be exported.",
+                    err
+                );
+                return Ok(SdkMeterProvider::builder().build());
+            }
+        };
+
+        let interval = export_interval.unwrap_or_else(|| {
+            Duration::from_millis(get_metric_export_interval_ms(&config) as u64)
+        });
+
+        let telemetry_exporter = TelemetryTrackingExporter::new(exporter, protocol);
+
+        let reader = opentelemetry_sdk::metrics::PeriodicReader::builder(telemetry_exporter)
+            .with_interval(interval)
+            .build();
+
+        let mut resource_attrs: Vec<opentelemetry::KeyValue> = Vec::new();
```

Comment:
Could you isolate all dd config -> otel resource manipulation to a separate function? I think it'd be easier to read

Reply:
I moved this logic to it's own method but I kept this logic in the metric reader (since this is the only place it's currently used). 

In a follow up PR I can move this logic to it's own module. Datadog SDKs should support the `OTEL_RESOURCE_ATTRIBUTES` configuration for tracing (and eventually logs). Here's a doc with all the configurations we need to support: https://docs.google.com/document/d/1gtdjhsMSIvuTnM9E4EreI7oZ-bAUoAxkhXX0-fBvdDY/edit?tab=t.0#heading=h.rnd972k0hiye